sys/include/net/utils.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_utils   Network helper functions
+ * @ingroup     net
+ * @brief       Common network helper functions
+ * @{
+ *
+ * @file
+ * @brief       Common network interface API definitions
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#ifndef NET_UTILS_H
+#define NET_UTILS_H
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include "net/ipv6/addr.h"
+#include "net/netif.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Parse an IPv6 address / hostname string.
+ *          If the @ref net_sock_dns module is used, this will
+ *          attempt to resolve hostnames via DNS.
```

Comment:
```suggestion
 *          attempt to resolve hostnames via DNS AAAA records.
```

or

```suggestion
 *          attempt to resolve hostnames via DNS to IPv6 addresses.
```

or

```suggestion
 *          attempt to resolve hostnames to IPv6 addresses
```

All should be clear from the context, but let's be either more or less specific for the uninitiated reader ;-).